### PR TITLE
fix(pages): publish only from main

### PR DESCRIPTION
**Pull request by saturley-hall**
## Why and what changed

The Pages workflow could not read the private site's configuration, and manual runs from other branches could reach publication steps. Keep site builds and tests available on pull requests while restricting artifact upload, configuration, and deployment to the exact `main` ref.

PR builds receive only `contents: read`, with checkout credential persistence disabled. Pages configuration runs immediately before deployment in the existing main-only deploy job; its `pages: write` grant covers both configuration reads and publication. `configure-pages` is pinned to the same verified upstream commit as its current `v5` tag (`983d7736d9b0ae728b81ab479565c72886d7745b`). The existing `github-pages` environment was verified to allow only the exact `main` branch.

## Review map

- Risk level: low; workflow permissions and publication routing.
- Start with `.github/workflows/pages.yml`: build permissions, artifact-upload guard, and deploy-job guard/permissions.
- Public or serialized contract changed: no.
- Compatibility or rollback concern: PR validation remains enabled; manual non-main runs build and test without publishing. Reverting would restore the private-site configuration failure and broader publication path.

## Evidence

Reviewed commit: `273562e9b09efb266e21230203cb735280b371ea`.

- `python3 scripts/build_pages_site.py --output-dir /private/tmp/aisimulate-pr166-site`: passed; 31 public files generated.
- `python3 -m unittest discover -s tests -p test_pages_site.py`: passed; 3 tests.
- `actionlint .github/workflows/pages.yml`: passed.
- Ruby YAML parsing and effective-permission checks: passed.
- `git diff --check`: passed.
- Fast CI: [passed on the reviewed head](https://github.com/ai-dynamo/aisimulate/actions/runs/34529570488). DCO and CODEOWNERS also passed.
- Full CI: [passed on the exact reviewed commit](https://github.com/ai-dynamo/aisimulate/actions/runs/34529590650): 18 successful jobs, with release staging intentionally skipped for manual PR validation.
- [Manual non-main Pages run](https://github.com/ai-dynamo/aisimulate/actions/runs/34529593853): passed; site build and all 3 tests succeeded, with upload and deployment skipped. [PR Pages validation](https://github.com/ai-dynamo/aisimulate/actions/runs/34529570354) also passed.
- CodeRabbit reviewed commit: `273562e9b09efb266e21230203cb735280b371ea`; completed successfully. All 4 actionable threads are fixed and resolved, including the follow-up request to pin configure-pages.
- Current-main compatibility: clean merge against `6e81f1ff05b60149a7d9258a1e7fdbd52094b154`, producing tree `6fc3078069b8a00dbd7debde4d0d34b196f9f167`. Site generation, all 3 site tests, and actionlint passed on that combined tree; only the intended workflow differs from main.
- Codex reviewed commit: `273562e9b09efb266e21230203cb735280b371ea`; no remaining actionable findings.
- Negative or boundary cases: all 7 guard cases passed: PR merge ref, PR event with main ref, main push, non-main push, main dispatch, non-main dispatch, and tag named main. Build permissions exclude Pages access; configuration is in deploy before publication.
- Expected-value derivation: N/A; no numerical behavior changed.
- Before/after: configuration moved from the PR-capable build job to the main-only deploy job; all 4 review threads answered and resolved.

## Modeling or data provenance

N/A; no model, dataset, or public API changes.

## Tracking

- Closes: N/A.
- Related PRs or issues: N/A; follow-up addresses the review threads on this PR.

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ on:
 
 permissions:
   contents: read
+  # actions/configure-pages queries the private Pages site configuration.
+  pages: read
```
Comment (jasonqinzhou) [inline .github/workflows/pages.yml]: pages: read is workflow-scoped here, so the pull_request build inherits it. Because checkout persists the GITHUB_TOKEN credential by default and the following steps execute PR-controlled Python, pull-request code can use that token to query the repository's Pages configuration even though only publication needs this access. Please leave the workflow/build job at contents: read and isolate Pages access to a main-only job—for example, move Configure GitHub Pages into the existing deploy job, which already has pages: write, while leaving the guarded artifact upload in build. This preserves PR site validation without granting it Pages access.

**Review by jasonqinzhou — APPROVED** [review]
**APPROVE · 8.2/10 · high confidence**

## What this PR does

This PR restricts GitHub Pages configuration, artifact upload, and deployment to the exact main branch while retaining pull-request builds and site-contract tests, and it adds the Pages API permission needed by configure-pages.

The publication guard is applied consistently across configuration, upload, and deployment, and the PR preserves pull-request validation instead of disabling the workflow wholesale.

## Why this score

The branch-gating change is small and functionally sound, and the targeted local checks pass. One reachable permission-boundary issue merits a non-blocking comment: the workflow-level pages read grant is inherited by pull-request jobs that execute PR-controlled code, although only the main publication path needs that access.

File: .github/workflows/pages.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/pages.yml]: Fixed in 2140eec68c8285b10a6f725fa664f9fb99d344c8. The workflow/build token now has only contents: read, and checkout uses persist-credentials: false. Configure GitHub Pages now runs inside the existing main-only deploy job, whose pages: write permission covers the configuration read. Artifact upload retains the same main-only guard. Local site generation, all 3 site-contract tests, actionlint, permission checks, and 7 branch/event boundary cases passed.

File: .github/workflows/pages.yml [hunk 2]
```diff
@@ -45,16 +47,22 @@ jobs:
       - name: Test public artifact contract
         run: python -m unittest discover -s tests -p test_pages_site.py
       - name: Configure GitHub Pages
-        if: github.event_name != 'pull_request'
+        if: >-
+          github.event_name != 'pull_request' &&
+          github.ref == 'refs/heads/main'
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/pages.yml]: Fixed in 2140eec68c8285b10a6f725fa664f9fb99d344c8. The workflow/build token now has only contents: read, and checkout uses persist-credentials: false. Configure GitHub Pages now runs inside the existing main-only deploy job, whose pages: write permission covers the configuration read. Artifact upload retains the same main-only guard. Local site generation, all 3 site-contract tests, actionlint, permission checks, and 7 branch/event boundary cases passed.

File: .github/workflows/pages.yml (hunk 1 above)
Comment (jasonqinzhou) [reply .github/workflows/pages.yml]: Fixed in 2140eec68c8285b10a6f725fa664f9fb99d344c8. The workflow/build token now has only contents: read, and checkout uses persist-credentials: false. Configure GitHub Pages now runs inside the existing main-only deploy job, whose pages: write permission covers the configuration read. Artifact upload retains the same main-only guard. Local site generation, all 3 site-contract tests, actionlint, permission checks, and 7 branch/event boundary cases passed.

File: .github/workflows/pages.yml [hunk 3]
```diff
@@ -37,33 +38,40 @@ jobs:
...
+      # Read the private Pages configuration and publish the site.
       pages: write
+      # Authenticate the deployment to GitHub Pages.
       id-token: write
     environment:
       name: github-pages
       url: ${{ steps.deployment.outputs.page_url }}
     steps:
+      - name: Configure GitHub Pages
+        uses: actions/configure-pages@v5
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/pages.yml]: Fixed in 273562e9b09efb266e21230203cb735280b371ea. Configure GitHub Pages is now pinned to 983d7736d9b0ae728b81ab479565c72886d7745b, the verified upstream commit currently referenced by v5 and already used by the existing workflow. This preserves the action implementation while preventing future tag movement. actionlint, YAML parsing, and git diff --check passed.
